smarts/core/waypoints.py

```diff
@@ -290,22 +291,27 @@ def _equally_spaced_path(self, path, point):
         ref_waypoints_coordinates = {
             parameter: [] for parameter in (continuous_variables + discrete_variables)
         }
-        for waypoint in path:
-            ref_waypoints_coordinates["ref_wp_positions_x"].append(waypoint.wp.pos[0])
-            ref_waypoints_coordinates["ref_wp_positions_y"].append(waypoint.wp.pos[1])
-            ref_waypoints_coordinates["ref_wp_headings"].append(
-                waypoint.wp.heading.as_bullet
-            )
-            ref_waypoints_coordinates["ref_wp_lane_id"].append(waypoint.wp.lane_id)
-            ref_waypoints_coordinates["ref_wp_lane_index"].append(
-                waypoint.wp.lane_index
-            )
-            ref_waypoints_coordinates["ref_wp_lane_width"].append(
-                waypoint.wp.lane_width
-            )
-            ref_waypoints_coordinates["ref_wp_speed_limit"].append(
-                waypoint.wp.speed_limit
-            )
+        for idx, waypoint in enumerate(path):
+            if idx == 0 or idx == len(path) - 1 or waypoint.is_shape_wp == 1:
```

Comment:
```suggestion
            if not waypoint.is_shape_wp or idx > 0 and idx < len(path) - 1:
                continue
```

Reply:
Thanks, sure, but the code which you have attached has issues, according to the `De Morgan's laws` in the following link:

https://en.wikipedia.org/wiki/De_Morgan%27s_laws
The correct syntax would be something like this
```
if not waypoint.is_shape_wp and 0<idx<len(path) - 1:
                continue
```